Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Snapping/snap.h

```diff
@@ -0,0 +1,518 @@
+      const vertex_descriptor vb2 = it.first->second;
+
+      // If there are multiple candidates for a source vertex, keep the closest target vertex
+      if(vb2 != vb)
+      {
+        typename CGAL::cpp11::unordered_map<vertex_descriptor, FT>::iterator dist_it =
+            m_sq_distance_to_snapped_point.find(va);
+        CGAL_assertion(dist_it != m_sq_distance_to_snapped_point.end());
+
+        const FT sq_dist_to_prev_best = dist_it->second;
+        if(CGAL::compare(sq_dist, sq_dist_to_prev_best) == CGAL::SMALLER)
+        {
+          VCM_left_iterator hint = it.first;
+          ++hint;
+          m_vertex_map.left.erase(it.first);
+          m_vertex_map.left.insert(hint, std::make_pair(va, vb));
+          dist_it->second = sq_dist;
+        }
+      }
+      else
+      {
+        m_sq_distance_to_snapped_point[va] = sq_dist;
+      }
+    }
+  }
+
+private:
+  Vertex_correspondence_map& m_vertex_map;
+  CGAL::cpp11::unordered_map<vertex_descriptor/*source*/, FT> m_sq_distance_to_snapped_point;
+
+  const SVPM& svpm;
+  const TVPM& tvpm;
+  const ToleranceMap& tol_vm;
+  const GeomTraits& gt;
+};
+
+// \ingroup PMP_repairing_grp
+//
+// Attempts to snap the vertices in `source_vrange` onto the vertices in `target_vrange`.
+// A vertex of the source range is only snapped to a vertex of the target mesh
+// if its distance to the target mesh vertex is smaller than a user-chosen bound.
+// If any source vertex can be snapped onto multiple vertices of the target
+// range, the closest one is chosen.
+// If multiple vertices within the source range are to be snapped to the same target vertex,
+// then the snapping is not performed for these vertices.
+//
+// @tparam PolygonMesh a model of `FaceListGraph` and `MutableFaceGraph`
+// @tparam VertexRange a model of `Range` with value type `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+// @tparam ToleranceMap a model of `ReadablePropertyMap` with key type `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+//                      and value type `GetGeomTraits<PolygonMesh, SourceNamedParameters>::type::FT`
+// @tparam SourceNamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+// @tparam TargetNamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+//
+// @param source_vrange a range of vertices of the source mesh whose positions can be changed.
+//                      the vertices must be border vertices of `smesh`.
+// @param smesh the source mesh whose border vertices might be moved
+// @param target_vrange a range of vertices of the target mesh which are potential new positions
+//                      for the vertices in the source range
+// @param tmesh the target mesh to which the vertices in `target_vrange` belong
+// @param tol_vm a tolerance map associating to each vertex of the source range a tolerance radius:
+//               potential projection targets are sought in a box centered on that vertex, and whose
```

Comment:
is it really a box? in my initial version, it was a search (there was a distance test after the bbox filtering)

Reply:
No, it of course checks against the proper 'eps' distance. Must have been a brain... snap:exclamation: